tests/gfx/TestLauncher.java

```diff
@@ -0,0 +1,55 @@
+package gfx;
+
+import java.io.*;
+import javax.microedition.io.file.*;
+import javax.microedition.io.*;
+import javax.microedition.midlet.*;
+import com.nokia.mid.impl.jms.core.*;
+
+public class TestLauncher extends MIDlet {
+    byte[] read(InputStream is) throws IOException {
+        int l = is.available();
+        byte[] buffer = new byte[l+1];
+        int length = 0;
+
+        while ((l = is.read(buffer, length, buffer.length - length)) != -1) {
+            length += l;
+            if (length == buffer.length) {
+                byte[] b = new byte[buffer.length + 4096];
+                System.arraycopy(buffer, 0, b, 0, length);
+                buffer = b;
+            }
+        }
+
+        return buffer;
+    }
```

Comment:
(Note to self: the next time I see one of these *read(InputStream)* methods, or similar methods like the *read(String)* method in #794, I should request that they get refactored into a utilities class. ;-) )